File: jmj-provider-api/jmj-provider-api-room/src/main/java/com/wllfengshu/jmj/provider/api/room/RoomService.java
package com.wllfengshu.jmj.provider.api.room;

import com.wllfengshu.jmj.provider.api.room.model.createroom.CreateRoomRequest;
import com.wllfengshu.jmj.provider.api.room.model.createroom.CreateRoomResponse;
import com.wllfengshu.jmj.provider.api.room.model.giveroom.GiveRoomRequest;
import com.wllfengshu.jmj.provider.api.room.model.giveroom.GiveRoomResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

/**
 * @author wangll
 * @date 2022-02-13 20:04
 */
@FeignClient(name = "jmj-provider-room")
public interface RoomService {

    @PostMapping(value = "/giveRoom")
    GiveRoomResponse giveRoom(@RequestBody GiveRoomRequest request);

    @PostMapping(value = "/createRoom")
    CreateRoomResponse createRoom(@RequestBody CreateRoomRequest request);
}

File: client/assets/scripts/server/RoomService.ts
import BaseService from "./BaseService";

export default class RoomService extends BaseService {
     private static readonly serverName : string = "jmj-provider-room";
     public static readonly instance : RoomService = new RoomService();

     // 获取房间
     public giveRoom(request : GiveRoomRequest) : GiveRoomResponse {
          return <GiveRoomResponse> JSON.parse(super.callInterface(RoomService.serverName, "giveRoom", request));
     }

     // 创建房间
     public createRoom(request : CreateRoomRequest) : CreateRoomResponse {
          return <CreateRoomResponse> JSON.parse(super.callInterface(RoomService.serverName, "createRoom", request));
     }

     // 加入房间
     public joinRoom(request : JoinRoomRequest) : JoinRoomResponse {
          return <JoinRoomResponse> JSON.parse(super.callInterface(RoomService.serverName, "joinRoom", request));
     }

}

